samples/react/counter-list/app.fs

```diff
@@ -0,0 +1,76 @@
+module App
+
+(**
+ - title: Counter-List
+ - tagline: The famous Increment/Decrement list sample ported from Elm
+*)
+
+open Fable.Core
+open Fable.Import
+open Elmish
+
+// MODEL
+
+type ID = int
+
+type Model = {
+    Counters : Counter.Model list
+}
+
+type Msg = 
+| Insert
+| Remove
+| Modify of ID * Counter.Msg
+
+let init() : Model =
+    { Counters = [] }
+
+// UPDATE
+
+let update (msg:Msg) (model:Model) =
+
+    match msg with
+    | Insert ->
+        let newCounter = Counter.init()
+        { Counters = newCounter :: model.Counters }
+
+    | Remove ->
+        { Counters = List.tail model.Counters }
+
+    | Modify (id, counterMsg) ->
+        { model with
+            Counters =
+                model.Counters
+                |> List.mapi (fun i counterModel -> 
+                    if i = id then
+                        Counter.update counterMsg counterModel
+                    else
+                        counterModel) }
+
+open Fable.Core.JsInterop
+open Fable.Helpers.React.Props
+module R = Fable.Helpers.React
+
+// VIEW (rendered with React)
+
+let view model dispatch =
+    let remove = R.button [ OnClick (fun _ -> dispatch Remove) ] [ R.str "Remove" ]
+
+    let insert = R.button [ OnClick (fun _ -> dispatch Insert) ] [ R.str "Add" ]
+
+    let counterDispatch i msg = dispatch (Modify (i, msg))
```

Comment:
Elm avoids explicit `dispatch` arg in the view by providing all the HTML properties that can result in a message as functions that return one (`fun 'a -> 'msg` instead of `fun 'a -> unit`) .